src/frequenz/dispatch/_dispatcher.py

```diff
@@ -213,6 +213,10 @@ async def start(self) -> None:
         """Start the actor."""
         self._actor.start()
 
+    async def refresh(self) -> None:
+        """Re-fetch all dispatches."""
+        await self._actor.fetch()
```

Comment:
I don't think this is a good idea Design-wise. The idea is to only communicate with actors via channels once they are created. Maybe for this case is overkill, at least until we can really distribute actors in other processes/nodes, and if that is the case ok, but I would leave it as a private method and just add an ignore and a comment saying this is a hack.

Reply:
Right thanks! My use case:
I have an actor that uses Dispacher.
When Dispacher starts (For the first time) it sends all dispaches and I receive them.

If my actor crashes and restarts latest dispaches are not send. And this is understandable because dispacher doesn't know that I restarted.
Of course I can create new dispacher on every restart. Would it be ok?

Reply:
Just to clarify, as you commented in this review comment, I'm only referring to the call to the actor inside `refresh()`, adding `refresh()` itself is OK because `Dispatcher` is just a class that uses an actor underneath (and the class instance should always be *local*, only the actor might be in another process/node).

That said, yeah, I think if it crashes it should be OK to re-create, it shouldn't happen all the time anyway.

**But** I don't oppose to this addition, I think things could go wrong and having a way to force a refresh could still be useful in some cases. But maybe we should add some sort of `Warning` if `refresh()` is not really intended to be used in normal conditions, just to avoid confusion or people thinking they need to actively refresh the dispatcher instance.

Reply:
BTW, the dispatcher should be instantiated by the app, not an actor, right? We should have only one dispatcher for the whole app/all actors, no?

Reply:
Ok, I will think and redesign my use case... Sorry